ttnn/cpp/ttnn/tensor/tensor_impl.cpp

```diff
@@ -686,21 +686,21 @@ Tensor to_host(const Tensor& tensor, bool blocking) {
     }
 }
 
-template Tensor to_host<bfloat16>(const Tensor& tensor, bool blocking);
-template Tensor to_host<float>(const Tensor& tensor, bool blocking);
-template Tensor to_host<int32_t>(const Tensor& tensor, bool blocking);
-template Tensor to_host<uint32_t>(const Tensor& tensor, bool blocking);
-template Tensor to_host<uint16_t>(const Tensor& tensor, bool blocking);
-template Tensor to_host<uint8_t>(const Tensor& tensor, bool blocking);
+template Tensor to_host<bfloat16>(const Tensor& tensor, bool blocking, uint8_t cq_id);
+template Tensor to_host<float>(const Tensor& tensor, bool blocking, uint8_t cq_id);
+template Tensor to_host<int32_t>(const Tensor& tensor, bool blocking, uint8_t cq_id);
+template Tensor to_host<uint32_t>(const Tensor& tensor, bool blocking, uint8_t cq_id);
+template Tensor to_host<uint16_t>(const Tensor& tensor, bool blocking, uint8_t cq_id);
+template Tensor to_host<uint8_t>(const Tensor& tensor, bool blocking, uint8_t cq_id);
 
 template <>
-Tensor to_host<bfloat4_b>(const Tensor& tensor, bool blocking) {
-    return to_host<uint32_t>(tensor, blocking);
+Tensor to_host<bfloat4_b>(const Tensor& tensor, bool blocking, uint8_t cq_id) {
+    return to_host<uint32_t>(tensor, blocking, cq_id);
 }
 
 template <>
-Tensor to_host<bfloat8_b>(const Tensor& tensor, bool blocking) {
-    return to_host<uint32_t>(tensor, blocking);
+Tensor to_host<bfloat8_b>(const Tensor& tensor, bool blocking, uint8_t cq_id) {
```

Comment:
why do we need all this templates if they are always calling to_host<uint32_t> anyway?

Reply:
I didn't write this code, just added the `cq_id` parameter to it. I would assume its to ensure that we have explicit and type-safe behaviour.